src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -91,6 +92,7 @@
     if (length(x[[idx]]) > 0 && length(x[[idx]][1]) == 1)
     {
       val <- x[[idx]][1]
+      col_type_r <- typeof(val)
```

Comment:
We should discuss this with @hadley. I think that several different column types from the perspective of `readr` may map to what appears to be the same type in R (for instance, the `col_number` type parses into vectors of type `double`). That implies that we might need to change the UX a little to show "parse as" in addition to or instead of the inferred data type of the result.